engine/src/main/java/org/terasology/logic/characters/interactions/EditScreenComponent.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.characters.interactions;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.module.sandbox.API;
+
+/**
+ * Entities with this component will show a different UI Screen during interactions when edit mode is on. <br/>
+ *
+ */
+@API
+public class EditScreenComponent implements Component {
```

Comment:
Rather than the new variable on `CharacterComponent` should having this component attached instead be the trigger for whether something can edit *at all* ? If no component then nope no edit for you. If there is a component variables here can customize it further if needed